toolchain/check/check.cpp

```diff
@@ -27,13 +27,18 @@ using ImportKey = std::pair<llvm::StringRef, llvm::StringRef>;
 // imports, not the main package declaration; as a consequence, it will be
 // `None` for the main package declaration.
 static auto GetImportKey(UnitAndImports& unit_info,
-                         IdentifierId file_package_id,
+                         PackageNameId file_package_id,
                          Parse::Tree::PackagingNames names) -> ImportKey {
   auto* stores = unit_info.unit->value_stores;
+  PackageNameId package_id =
+      names.package_id.has_value() ? names.package_id : file_package_id;
+  auto package_ident_id = package_id.AsIdentifierId();
   llvm::StringRef package_name =
-      names.package_id.has_value() ? stores->identifiers().Get(names.package_id)
-      : file_package_id.has_value() ? stores->identifiers().Get(file_package_id)
-                                    : "";
+      package_id.has_value()
+          ? (package_ident_id.has_value()
+                 ? stores->identifiers().Get(package_ident_id)
+                 : package_id.AsSpecialName())
+          : "";
```

Comment:
I'd wonder if this would be easier to read with an `if` for the outer ternary, avoiding `package_ident_id` when invalid:

```
  llvm::StringRef package_name;
  if (package_id.has_value()) {
    auto package_ident_id = package_id.AsIdentifierId();
    package_name = package_ident_id.has_value() ?  package_ident_id.AsSpecialName() : stores->identifiers().Get(package_ident_id);
  }
```

Thoughts?

Reply:
That looks nicer to me, done.